src/components/Pages/WithPlayer/widgets/UnitMaterials/Summary/Summary.js

```diff
@@ -12,35 +12,40 @@ import { selectors as assetsSelectors, doc2html } from '../../../../../../redux/
 import { INSERT_TYPE_SUMMARY } from '../../../../../../helpers/consts';
 import MenuLanguageSelector from '../../../../../../components/Language/Selector/MenuLanguageSelector';
 
-const Summary = ({ unit, t }) => {
+export const getSummaryLanguages = unit =>
+  (unit && unit.files &&
+    unit.files.filter(f =>
+      MediaHelper.IsText(f) && !MediaHelper.IsPDF(f) && f.insert_type === INSERT_TYPE_SUMMARY)
+      .map(f => f.language)) || [];
+
+export const getFile = (unit, lang) => {
+  if (!unit || !Array.isArray(unit.files)) {
+    return null;
+  }
+
+  return unit.files?.filter(f => f.language === lang)
+    .filter(f => MediaHelper.IsText(f) && !MediaHelper.IsPDF(f))
+    .find(f => f.insert_type === INSERT_TYPE_SUMMARY);
```

Comment:
These 2 lines are the same as 17-18. Maybe to make it a function with a meaningful name?

Reply:
They are not the same.

First will "getAllSummaryFiles" and the one you commented on will "getFirstSummaryFileOrNull".
I tried refactoring this into "getAllTextFiles" and then getting the first or filtering summary (again) which does not result in a nice code... so I think we should keep this as is.